src/mon/AuthMonitor.cc

```diff
@@ -1795,6 +1795,47 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
       + " tag " + pg_pool_t::APPLICATION_NAME_CEPHFS
       + " data=" + filesystem;
 
+    vector<string> newcaps = {"mon", mon_cap_string, "osd", osd_cap_string,
+			      "mds", mds_cap_string};
+
+    EntityAuth e_auth;
+    if (mon.key_server.get_auth(entity, e_auth)) {
+      for (auto it = newcaps.begin(); it != newcaps.end(); it += 2) {
```

Comment:
This loop has been removed in commit 0e445f84616d812b6c2e3236d07c5b535d5cbc6b - why not just remove it in this commit?

Reply:
The loop wasn't added, it was just modified. 0e445f8 deals with a change different from c0d4aa8.

Reply:
IMO, it the commits should be squashed. Can you explain why they are entirely different?

Reply:
Please take a look at the commit messages.

Reply:
I already did, and it still looks pretty close enough to be treated as a single commit:

AuthMonitor: allow "fs authorize" to update caps
AuthMonitor: update caps when no caps are present in keyring

Reply:
> AuthMonitor: allow "fs authorize" to update caps

This commit adds the ability to add caps using `fs authorize` command

> AuthMonitor: update caps when no caps are present in keyring

And this commit changes the rule that empty keyring cannot be updated.

Reply:
@vshankar ping.

Reply:
It still looks related, however, I'm not going to hold the progress for this change - so, I'm fine with having this as separate commits.